hack/local-development/start-extension-provider-local

```diff
@@ -52,7 +52,7 @@ if [ "$USER" != root ]; then
   SUDO="sudo -E"
 fi
 
-$SUDO PROVIDER_LOCAL_DISABLE_COREDNS_MUTATION=true go run \
+PROVIDER_LOCAL_DISABLE_COREDNS_MUTATION=true $SUDO go run \
```

Comment:
For similarity with other env vars that are configured for provider-local (`LEADER_ELECTION_NAMESPACE`, `GARDENER_SHOOT_CLIENT `) can we move this env var to https://github.com/gardener/gardener/blob/35333c9f75b15afc493612eb9fcd9132a1e80bab/hack/local-development/start-extension-provider-local#L37-L39?